gustaf/helpers/notebook.py

```diff
@@ -0,0 +1,136 @@
+        (nx - 1, ny - 1),
+    ]
+    ind, minl = 0, 1000
+    for i, m in enumerate(lm):
+        l_something = m[0] * m[1]
+        if N <= l_something < minl:
+            ind = i
+            minl = l_something
+    return lm[ind]
+
+
+class K3DPlotterN(GridspecLayout):
+    """Helper to plot in notebooks with k3d.
+
+    Sets up K3D plotter especially if multiple plots are to be shown in the
+    notebook.
+    """
+
+    def __init__(self, N, size, background=0xFFFFFF):
+        """Setups the plotter with potentially multiple plots.
+
+        Parameters
+        ----------
+        N: int
+            Number of plots to be shown.
+        size: (str, list)
+            See vedo.Plotter for details.
+        background: ((color, str), optional)
+          See vedo.Plotter for details. Defaults to 0xFFFFFF.
+        """
+        if importlib.util.find_spec("k3d") is None:
+            raise ImportError(
+                "k3d is not installed. Please install it with `pip "
+                "install k3d`."
+            )
+        self.N = N
+        self.x, self.y = get_shape(N, *(2160, 1440))
+        self.shape = (self.x, self.y)
+        super().__init__(self.x, self.y)
+        self.vedo_plotters = []
+        for _ in range(N):
+            self.vedo_plotters.append(
+                vedo.Plotter(
+                    size=size,
+                    bg=background,
+                )
+            )
+
+    def _at_get_location(self, N):
+        """Gets the grid coordinate of the wanted renderer.
+
+        Parameters
+        ----------
+        N: int
+            Render id.
+
+        Returns:
+            (int, int): Grid Coordinate of the wanted renderer.
+        """
+        if (self.x * self.y) < N:
+            return (self.x - 1, self.y - 1)
+        return (N // (self.y), N % self.y)
+
+    def show(self, list_of_showables, at, interactive, camera, axes):
+        """Add the showables to the renderer at the given location.
+
+        Parameters
+        -----------
+        list_of_showables: Any
+        at: int
+            Render id.
+        interactive: bool
+            See vedo.Plotter.show for details.
+        camera: Any
+            See vedo.Plotter.show for details.
+        axes: bool
+            Add axes to the plot. Will also cast int to bool.
+        """
+        self[self._at_get_location(at)] = self.vedo_plotters[at].show(
+            list_of_showables,
+            interactive=interactive,
+            camera=camera,
+            axes=axes,
+            # offscreen=offscreen,
+        )
+
+    def display(self):
+        """Display the plotter.
+
+        This is needed in case the plotter is the last thing in a cell. In that
+        case the IPython will try to call this function to display this.
+        """
+        display(self)
+
+    def clear(self, *args, **kwargs):
+        """Clear the plotters."""
+        for renderer in self.vedo_plotters:
+            renderer.clear(*args, deep=True, **kwargs)
```

Comment:
```suggestion
            renderer.clear(deep=True)
```
do you still get error with this?

Reply:
Not applicable